gulpfile.js

```diff
@@ -770,3 +770,55 @@ gulp.task('release', gulp.series(['build', 'typings', function() {
 
 // The default task builds Blockly.
 gulp.task('default', gulp.series(['build']));
+
+
+// Stash current state, check out the named branch, and sync with
+// google/blockly.
+function syncBranch(branchName) {
+  return function(done) {
+    execSync('git stash save -m "Stash for sync"', { stdio: 'inherit' });
+    execSync('git checkout ' + branchName, { stdio: 'inherit' });
+    execSync('git pull https://github.com/google/blockly.git ' + branchName,
+        { stdio: 'inherit' });
+    execSync('git push origin ' + branchName, { stdio: 'inherit' });
+    done();
+  }
+}
+
+// Stash current state, check out develop, and sync with google/blockly.
+gulp.task('git-sync-develop', syncBranch('develop'));
+
+// Stash current state, check out master, and sync with google/blockly.
+gulp.task('git-sync-master', syncBranch('master'));
+
+// Helper function: get a name for a rebuild branch. Format: rebuild_mm_dd_yyyy.
+function getRebuildBranchName() {
+  var date = new Date();
+  var mm = date.getMonth() + 1; // Month, 0-11
+  var dd = date.getDate(); // Day of the month, 1-31
+  var yyyy = date.getFullYear();
+  return 'rebuild_' + mm + '_' + dd + '_' + yyyy;
+};
+
+// Recompile and push to origin.
+gulp.task('recompile', gulp.series([
+    'git-sync-develop',
+    function(done) {
+      execSync('git stash save -m "Stash for rebuild"', { stdio: 'inherit' });
```

Comment:
At this point we've already stashed with git-sync-develop, any need to stash again?

Reply:
Nope, removed.